protostar-rust/src/cheatcodes_hint_processor.rs

```diff
@@ -202,32 +224,68 @@ fn call_contract(
     Ok(return_data)
 }
 
-#[allow(unused, clippy::too_many_lines)]
 fn execute_cheatcode_hint(
     vm: &mut VirtualMachine,
-    exec_scopes: &mut ExecutionScopes,
-    hint: &ProtostarHint,
+    _exec_scopes: &mut ExecutionScopes,
+    blockifier_state: &mut CachedState<DictStateReader>,
+    selector: &BigIntAsHex,
+    [input_start, input_end]: [&ResOperand; 2],
+    [output_start, output_end]: [&CellRef; 2],
+) -> Result<(), HintError> {
+    // Parse the selector.
+    let selector = &selector.value.to_bytes_be().1;
+    let selector = std::str::from_utf8(selector).map_err(|_| {
+        HintError::CustomHint(Box::from(
+            "Failed to parse the  cheatcode selector".to_string(),
+        ))
+    })?;
+
+    // Extract the inputs.
+    let input_start = extract_relocatable(vm, input_start)?;
+    let input_end = extract_relocatable(vm, input_end)?;
+    let inputs = read_data_from_range(vm, input_start, input_end)
+        .map_err(|_| HintError::CustomHint(Box::from("Failed to read input data".to_string())))?;
+
+    match_cheatcode_by_selector(
+        vm,
+        blockifier_state,
+        selector,
+        inputs,
+        [output_start, output_end],
+    )
+}
+
+#[allow(unused, clippy::too_many_lines)]
```

Comment:
why is `unused` here?

Reply:
We do not currently use `contract_address` from the `deploy` cheatcode, there is an issue to check the address afaik